File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 3
end

# :nodoc
class Hamming
  def self.compute(a, b)
    raise ArgumentError, 'Lengths don\'t match' if a.length != b.length
    a.split('').each.with_index.inject(0) do |sum, (e, i)|
      sum + (e == b[i] ? 0 : 1)
    end
  end
end

File: go/hamming/hamming.go
// Package hamming does a thing
package hamming

import "fmt"

// Distance does a thing
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, fmt.Errorf("string lengths don't match")
	}
	distance := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}
	return distance, nil
}
